### PR TITLE
Syntax fix in Simplified Likelihoods

### PR DESCRIPTION
**Pull request by jackaraz**
This PR fixes some syntax issues with likelihood functions

### CODE REVIEW

File: smodels/tools/simplifiedLikelihoods.py [hunk 1]
```diff
@@ -1077,23 +1098,25 @@ def _ul_preprocess(
         if signal_type == "signal_rel":
             sigma_mu = computer.getSigmaMu(mu_hat/sum(model.nsignal), theta_hat0)
             sigma_mu = sigma_mu * sum(model.nsignal)
+            # TODO convert rel_signals to signals
```
Comment (jackaraz) [inline smodels/tools/simplifiedLikelihoods.py:1101]: `rel_signals` needs to be converted to `signals`

File: smodels/tools/simplifiedLikelihoods.py [hunk 2]
```diff
@@ -1077,23 +1098,25 @@ def _ul_preprocess(
...
-                nll=True, mu = mu_hat,
             )
         # print ( f"SL nll0 {nll0:.3f} muhat {mu_hat:.3f} sigma_mu {sigma_mu:.3f}" )
         if np.isinf(nll0) and not marginalize and not trylasttime:
             logger.warning(
                 "nll is infinite in profiling! we switch to marginalization, but only for this one!"
             )
             marginalize = True
+            # TODO convert rel_signals to signals
             nll0 = computer.likelihood(
```
Comment (jackaraz) [inline smodels/tools/simplifiedLikelihoods.py:1120]: `rel_signals` needs to be converted to `signals`

File: smodels/tools/simplifiedLikelihoods.py [hunk 3]
```diff
@@ -1113,6 +1136,7 @@ def _ul_preprocess(
         mu_hatA = compA.findMuHat()
         if signal_type == "signal_rel":
             mu_hatA = mu_hatA * sum ( aModel.nsignal)
+        # TODO convert rel_signals to signals
         nll0A = compA.likelihood(
```
Comment (jackaraz) [inline smodels/tools/simplifiedLikelihoods.py:1140]: `rel_signals` needs to be converted to `signals`

File: smodels/tools/simplifiedLikelihoods.py [hunk 4]
```diff
@@ -1131,6 +1155,7 @@ def clsRoot(mu: float, return_type: Text = "CLs-alpha") -> float:
                         CLs: returns CLs value
             """
             ## the function to find the zero of (ie CLs - alpha)
+            # TODO convert rel_signals to signals
             nsig = getattr(model, "rel_signals" if signal_type == "signal_rel" else "nsignals")(mu)
```
Comment (jackaraz) [inline smodels/tools/simplifiedLikelihoods.py:1159]: `rel_signals` needs to be converted to `signals`

File: smodels/tools/simplifiedLikelihoods.py [hunk 5]
```diff
@@ -963,11 +982,13 @@ def chi2(self, marginalize=False):
...
-        llhd = self.likelihood(nsig, marginalize=marginalize, nll=True, mu = 1. )
+        llhd = self.likelihood_beta(1., marginalize=marginalize, nll=True)
 
         # Compute the maximum likelihood H1, which sits at nsig = nobs - nb
         # (keeping the same % error on signal):
         if len(nsig) == 1:
+            # TODO this nsig initiation seems wrong and changing maxllhd to likelihood_beta
+            # fails ./testStatistics.py : zero division error in L115
             nsig = self.model.observed - self.model.backgrounds
             maxllhd = self.likelihood(nsig, marginalize=marginalize, nll=True, mu = 1. )
```
Comment (jackaraz) [inline smodels/tools/simplifiedLikelihoods.py:993]: There is a problem with this `nsig` initiation. `likelihood_beta` fails the `testStatistics.py`.

File: smodels/tools/simplifiedLikelihoods.py [hunk 6]
```diff
@@ -891,6 +894,22 @@ def likelihood(self, nsig, marginalize=False, nll=False, mu=None):
         else:
             return self.profileLikelihood(nsig, nll)
 
+    def likelihood_beta(self, mu: float, marginalize: bool = False, nll: bool = False) -> float:
```
Comment (jackaraz) [inline smodels/tools/simplifiedLikelihoods.py:897]: @WolfgangWaltenberger once you are happy we can move beta to likelihood.

**Review by jackaraz — COMMENTED** [review]
Regarding the code structure

File: smodels/tools/simplifiedLikelihoods.py (hunk 6 above)
Comment (WolfgangWaltenberger) [reply smodels/tools/simplifiedLikelihoods.py:897]: @jackaraz unittests run through for me, am happy with the commit, only comment i have is about the comment:
simplifiedLikelihoods.py:898
compute likelihood for nsig, profiling the nuisances    -> compute likelihood for mu, profiling or marginalizing the nuisances

**Pull request merged**